TrackerGeom/inc/Tracker.hh

```diff
@@ -22,112 +21,67 @@
 
 #include "Mu2eInterfaces/inc/Detector.hh"
 #include "Mu2eInterfaces/inc/ProditionsEntity.hh"
-#include "TrackerGeom/inc/Manifold.hh"
-#include "TrackerGeom/inc/Support.hh"
-#include "TrackerGeom/inc/SupportModel.hh"
-#include "TrackerGeom/inc/SupportStructure.hh"
-#include "GeneralUtilities/inc/HepTransform.hh"
 
 #include "TrackerGeom/inc/Plane.hh"
 #include "TrackerGeom/inc/Panel.hh"
-#include "GeomPrimitives/inc/TubsParams.hh"
-#include "GeomPrimitives/inc/PlacedTubs.hh"
-#include "TrackerGeom/inc/PanelEB.hh"
 #include "TrackerGeom/inc/StrawProperties.hh"
+#include "TrackerGeom/inc/G4Tracker.hh"
 
 namespace mu2e {
   class Tracker : public Detector, public ProditionsEntity {
-    friend class TrackerMaker; // remove after factorizing out the G4 stuff FIXME!
-    using xyzVec = CLHEP::Hep3Vector; // switch to XYZVec TODO
-
+    using xyzVec = CLHEP::Hep3Vector; // switch to root XYZVec TODO
     public:
     typedef std::shared_ptr<Tracker> ptr_t;
     typedef std::shared_ptr<const Tracker> cptr_t;
+    using G4TrackerPtr = std::shared_ptr<G4Tracker>;
     using PlaneCollection = std::array<Plane,StrawId::_nplanes>;
     using PanelCollection = std::array<Panel,StrawId::_nupanels>;
     using StrawCollection = std::array<Straw,StrawId::_nustraws>;
-    using StrawIndexMap = std::array<uint16_t,StrawId::_maxval>; 
+    using StrawIndexMap = std::array<uint16_t,StrawId::_maxval>;
+
+    using PEType = std::array<bool,StrawId::_nplanes>;
+
     // default constructor results in non-functional object, but is required by proditions service
 
     constexpr static const char* cxname = {"Tracker"};
     // copy constructor
     Tracker(const Tracker& other);
     // construct from a set of straws and their global properties.  It would be better to take ownership of
     // existing straws via std::move, but the need for the copy constructor above precludes that
```

Comment:
I don't think that std::array supports std::move.  When you create the StrawCollection inside TrackerMaker it's made on the heap.  When you create the Tracker object, it is made on the heap and the memory for the StrawCollection is inside the Tracker object.  So there is no way to avoid the copy.

Also, I think that the compiler written copy c'tor will do the right thing.  So we should just use it rather than writing our own.